pkg/queue/handlers/http_request.go

```diff
@@ -0,0 +1,258 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/contiamo/go-base/pkg/queue"
+	"github.com/contiamo/go-base/pkg/queue/workers"
+	"github.com/contiamo/go-base/pkg/tokens"
+	"github.com/contiamo/go-base/pkg/tracing"
+	"github.com/opentracing/opentracing-go"
+	otext "github.com/opentracing/opentracing-go/ext"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// HTTPRequestTask marks a task as an HTTP request task
+	HTTPRequestTask queue.TaskType = "http-request"
+)
+
+// HTTPRequestTaskSpec describes the specification of the HTTP request task
+type HTTPRequestTaskSpec struct {
+	// Method to use for the HTTP request
+	Method string `json:"method"`
+	// URL is the target URL for the request.
+	// Must be an absolute URL that contains the scheme and the host components.
+	URL string `json:"url"`
+	// RequestBody to send
+	RequestBody string `json:"requestBody"`
```

Comment:
Perhaps it makes sense to use []byte for the request body and the ResponseBody below. That way the handler is also usable for cases where binary data is submitted that is not valid utf8

Reply:
this task handlers supports only `application/json` as a response, you can find the check in the code.